### DISCUSSION TITLE
Intellicenter suck on pool mode

### DISCUSSION THREAD

**Discussion by Tegguy** [tagyoureit/nodejs-poolController#566] [Q&A]
I don't know if this is a NJSPC issue or a Pentair issue but at the suggestion of @rstrouse I am posting it here as well. I've noticed a couple times now that when my scheduled pool run time ends for the day the "pool" will still show active. The pump will stop but the chlorinator will still have power and everything and I have to reboot the panel to fix this.

The pool status light will be green even though the pump isn't running and if I click to move to SPA mode the valves activate but the pool light doesn't seem to turn off.

I will probably need a little help creating a replay as I have never done this before.

Side question... I'm trying to run the git pull command to make sure I have the latest config but I get the error :
error: Your local changes to the following files would be overwritten by merge:
        package-lock.json
        web/bindings/mqtt.json
Please commit your changes or stash them before you merge.
Aborting

How can I fix this?

**Comment by rstrouse** [comment]
### Updating njsPC when git detects modified files

Unfortunately, git can be a bit fickle but the fix although cryptic is often very easy.  Run the following commands and it will reset the package-lock.json file.  If you made changes to your mqtt.json file, please save this file somewhere before running the commands below.  We can then define this as a custom interface and you will not have further issues with git.

`cd ~/nodejs-poolController`
`git reset --hard`
`git pull`
`npm i`
`cd ~/nodejs-poolController/dashPanel`
`git pull`
`nmp i`

After you have updated the software, upload your saved mqtt.json file by going to the gears menu and clicking on the Controller=>Interfaces tab.  Click Add Interface and upload your mqtt.json file by clicking the the Upload Bindings button.

Once the bindings are uploaded, select MQTT in the Type dropdown, name your new interface and select the file from the Binding dropdown.  Don't forget to disable your old MQTT interface and enable this new one you uploaded.  From here on out it will not affect your updates and preserve the file.

### After Updating your software


First you should capture a replay with your configuration.  Follow the instructions in the wiki and post the zip file here.  Then when you have this situation, run another capture.  This will tell us what IntelliCenter is up to.  My first suspicion is that the connectivity is spotty but we will know more when we see the replays.

https://github.com/tagyoureit/nodejs-poolController/wiki/How-to-capture-all-packets-for-issue-resolution

**Reply by Tegguy** [reply]
Thank you. I got the MQTT thing fixed and updated the code. I have a "standard" capture that I've attached and will capture the other capture when the issue happens again. This issue shows up on both NJSPC and the Intellicenter so I'm not sure if it's connectivity or not.

I did have another issue I was waiting to bring up but I am having issues controlling the iChlor... NJSPC would read the values but I couldn't change them or activate super chlorinate from the dash panel I had to goto the Intellicenter. Now that I've updated it's not even reading the iChlor is active.
[Capture Working Config.zip](https://github.com/tagyoureit/nodejs-poolController/files/9304297/Capture.Working.Config.zip)

**Reply by Tegguy** [reply]
Just an update... I got the iChlor values to show up but it still won't control anything

**Reply by rstrouse** [reply]
@Tegguy I need another replay of you trying to control things.  The iChlor in your configuration was not assigned to IntelliCenter.  When adding a chlorinator to IntelliCenter you need to select IntelliCenter as the Master for the chlorinator.  In the replay you have included, it is assigned as an externally controlled chlorinator.

**Reply by Tegguy** [reply]
@rstrouse I deleted the chlorinator and re-added it and this seems to have fixed the issue. Do you still need a new replay? I am still waiting for the equipment to have the original issue again showing the pool active with the pump off but so far it hasn't done it (thinking it's intermittent hardware issue at this point)

**Reply by rstrouse** [reply]
If it does it again capture a replay.  At that point we should be able to see what the panel is doing.

**Reply by Tegguy** [reply]
Just wanted to post an update... So far it has not done it again but will keep everyone posted
